### ISSUE TITLE
Cloudwatch Logs put_log_events Response Does Not Contain nextSequenceToken, Raises Exception in Watchtower

### ISSUE THREAD

**Issue by o-nikolas** [kislyuk/watchtower#134]
## Summary/Description of Issue
Using Watchtower versions 0.8.0 through 1.0.1 in our logs we are occasionally seeing this error (some portions redacted):

```
[2021-01-08 10:27:08,332] {my_logging.py:112} WARNING -   File "<python path>/python3.6/site-packages/watchtower/__init__.py", line 223, in emit
    self._submit_batch([cwl_message], stream_name)
[2021-01-08 10:27:08,386] {my_logging.py:112} WARNING -   File "<python path>/python3.6/site-packages/watchtower/__init__.py", line 191, in _submit_batch
    raise e
[2021-01-08 10:27:08,452] {my_logging.py:112} WARNING -   File <python path>/python3.6/site-packages/watchtower/__init__.py", line 186, in _submit_batch
    self.sequence_tokens[stream_name] = response["nextSequenceToken"]
[2021-01-08 10:27:08,509] {my_logging.py:112} WARNING - KeyError: 'nextSequenceToken'
``` 
An example response we're getting when this occurs looks like this (some portions redacted):
```
{'ResponseMetadata': {'RequestId': <request id>, 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': <request id>, 'content-type': 'application/x-amz-json-1.1', 'content-length': '80', 'date': 'Fri, 08 Jan 2021 18:27:07 GMT'}, 'RetryAttempts': 0}}
```
Which indeed does not contain a `nextSequenceToken`, but is otherwise a successful put log, we get a `200` back and the log is successfully written to the Cloudwatch Log stream. We can't determine the conditions under which Cloudwatch Logs does not send back a `nextSequenceToken`

## A Possible Fix

Updating the line which extracts the `nextSequenceToken` to look as follows solves the issue:
```python
if "nextSequenceToken" in response:                                  
    self.sequence_tokens[stream_name] = response["nextSequenceToken"]
```

This effectively leaves a stale sequence token for that stream in the cache, which is then caught during the next put and repaired by the existing mechanism to do so in Watchtower.

**Comment by kislyuk** [comment]
Thank you. This looks like a harmless defensive change so I went ahead and added it in.

**Comment by o-nikolas** [comment]
Thanks Andrey :)

**Comment by o-nikolas** [comment]
By the way, is it possible to get a minor release for this published?

**Comment by kislyuk** [comment]
Released in v1.0.5, please test.
